src/components/pages/EditorModals.jsx

```diff
@@ -27,6 +34,29 @@ const EditorModals = ({ mdeRef, onSave, modalType, onClose, mediaType }) => {
     onClose()
   }
 
+  const onInstagramEmbedSave = (type, value) => {
+    const cm = mdeRef.current.simpleMde.codemirror
+
+    if (type === "postUrl") {
+      // value should be of type { postUrl: string, isCaptioned: boolean }
+      const { postUrl, isCaptioned } = value
+      if (RegExp(INSTAGRAM_POST_URL_REGEX).test(postUrl)) {
```

Comment:
This would have been caught within InstagramModal as the Save button will be disabled if the regex test fails, so this test should definitely pass. If it doesn't then there's something wrong and we will not make any changes just in case.

Reply:
actually something like [branded types](https://dev.to/andersonjoseph/typescript-tip-safer-functions-with-branded-types-14o4) imght come in useful here, where the type can only be created by passsing the `RegExp` check. 

however, because this file is in `.jsx`, such guarantees aren't available here ;--; what _could_ happen is that because this **must** be true, we could throw an error if it fails. This would make the code clearer but runs the risk of errors being exposed ot the user (again, the invariant **must** be true for this approach and we should include a comment proving this invariant)